Dockerfile.dev

```diff
@@ -3,10 +3,28 @@ FROM python:3.6-alpine
 RUN apk update && apk add python3-dev build-base git
 
 WORKDIR /app
+
+# Some dependencies require a very long compilation: protobuf, numpy,
+# grpcio. To avoid having to re-install these packages every time, we
+# pre-install them so that they are cached.
+#
+# However, we cannot just pre-install a few packages, because we don't
+# know exactly which version `pip` will pick for them. Instead, we
+# give `pip` all the package's dependencies, and let it resolve and
+# install them.
 COPY setup.py .
-COPY xain_fl xain_fl/
+RUN mkdir xain_fl && \
+    printf '__version__ = "0"\n__short_version__ = "0"' > xain_fl/__version__.py && \
```

Comment:
Maybe I missed something in code review from before, but why do we need this dummy version here?

Reply:
Just so that `setup.py` can run. `setup.py` opens this file.